pandas/core/window/rolling.py

```diff
@@ -131,6 +132,32 @@ def __init__(
             )
         self.validate()
 
+    # TODO: Remove once win_type deprecation is enforced
+    def _shallow_copy(self, obj, **kwargs):
+        warnings.simplefilter("ignore", FutureWarning)
+        super()._shallow_copy(obj, **kwargs)
```

Comment:
```suggestion
        with warnings.catch_warnings():
            warnings.filterwarnings("ignore", "win_type", FutureWarning)
            super()._shallow_copy(obj, **kwargs)
```
this should ensure the warning filter is only changed inside that context
this should ensure